source/components/ThemeProvider.js

```diff
@@ -40,23 +40,6 @@ class ThemeProvider extends Component<Props, State> {
     };
   }
 
-  // checks if theme and/or themeOverrides props have changed
-  // in order to update state before rendering
-  componentWillReceiveProps(nextProps: Props) {
-    const { theme, themeOverrides } = nextProps;
-
-    const changedProps = _.pickBy(
-      { theme, themeOverrides },
-      (value, key) => this.props[key] !== value
-    );
-
-    if (Object.keys(changedProps).length > 0) {
-      this.setState({
-        theme: this.composeLibraryTheme(theme, themeOverrides)
-      });
-    }
-  }
-
```

Comment:
This is not necessary anymore?

Reply:
With the new context API only the children of consumers will re-render if context changes. It used to be necessary with the original version of context because the entire tree of children nested within a Provider would re-render if something changed in context.